src/VecSim/containers/raw_data_container_interface.h

```diff
@@ -41,4 +43,24 @@ struct RawDataContainer {
      * @return status
      */
     virtual RawDataContainer_Status updateElement(size_t id, const void *element) = 0;
+    /**
+     * Create a new iterator. Should be freed by the iterator's destroctor.
+     */
+    virtual RawDataContainer_Iterator *getIterator() = 0;
 };
+
+struct RawDataContainer_Iterator {
+    /**
+     * This is an abstract interface, constructor/destructor should be implemented by the derived
+     * classes
+     */
+    RawDataContainer_Iterator() = default;
+    virtual ~RawDataContainer_Iterator() = default;
+
+    /**
+     * The basic iterator operations API
+     */
+    virtual bool hasNext() = 0;
+    virtual const char *next() = 0;
+    virtual void reset() = 0;
```

Comment:
Any reason not to adapt to C++ API (`++` and `*` operators)?
Also consider making this class a sub-class of the main one

Reply:
I moved the iterator to be a sub-class. Regarding the operators, it's nice and sophisticated but doesn't make much difference... we already have iterators in this format which is clearer IMO.

Reply:
I don't think that a C API is clearer on a C++ project. I also don't like that we have both `hasNext` and `next` but `next` calls `hasNext` anyways